provision/kubernetes/provisioner.go

```diff
@@ -404,13 +404,21 @@ func podsForApps(client *ClusterClient, apps []provision.App) ([]apiv1.Pod, erro
 		}
 		sel = sel.Add(*req)
 	}
-	pods, err := client.CoreV1().Pods(client.Namespace()).List(metav1.ListOptions{
-		LabelSelector: sel.String(),
-	})
-	if err != nil {
-		return nil, err
+	poolNames := make([]string, len(apps))
```

Comment:
After messing with kubectl a bit I found that it's possible to list pods from all namespaces with a single request, I think this should be used here, the only catch is that apparently we have to create the request more manually, something like: 
```
restCli, err := rest.RESTClientFor(client.restConfig)
var podList apiv1.PodList
err = restCli.Get().
	Resource("pods").
	VersionedParams(&metav1.ListOptions{
<opts>
}, scheme.ParameterCodec).
	Do().
	Into(&podList)
```

Reply:
The problem is that this would list pods from all namespaces, not only the ones we use for pools - which include `kube-system`. This could be dangerous